sql/moz-fx-data-shared-prod/firefox_accounts_derived/fxa_users_first_seen_v1/query.sql

```diff
@@ -1,44 +1,37 @@
 WITH _current AS (
-  SELECT
+  SELECT DISTINCT
     user_id,
-    ARRAY_AGG(DISTINCT service IGNORE NULLS) AS services_used,
+    -- ARRAY_AGG(DISTINCT service IGNORE NULLS) AS services_used, # TODO: create an alternative agg model
+    --   for high-level user aggregated information for quick user overview.
   FROM
-    firefox_accounts.fxa_all_events
+    `firefox_accounts.fxa_users_daily`
   WHERE
-    DATE(`timestamp`) = @submission_date
-    AND event_category IN ('auth', 'auth_bounce', 'content', 'oauth')
-  GROUP BY
-    user_id
+    submission_date = @submission_date
+    -- ensuring we only process rows where user_id is provided
+    AND user_id IS NOT NULL
 ),
 _previous AS (
-  SELECT
-    *
+  SELECT DISTINCT
+    user_id
   FROM
-    firefox_accounts_derived.fxa_users_first_seen_v1
+    `firefox_accounts.fxa_users_first_seen`
   WHERE
     -- In reprocessing scenarios, we must always backfill from the first affected date
     -- all the way to the present; to enforce that, we explicitly drop any data after
     -- the target @submission_date
     first_seen_date < @submission_date
+    -- it appears we have a user_id will null value exist inside fxa_users_first_seen
+    -- excluding it here to avoid it being recreated for each new submission_date
+    AND user_id IS NOT NULL
```

Comment:
This is something I noticed in the current fxa_users_first_seen table, I assume this is a bug. What do you think @fbertsch ?

Reply:
IMO this should be dropped upstream. Not bad to also include it here though.

Reply:
As far as I can see we drop it now:
https://github.com/mozilla/bigquery-etl/blob/945c4ee0301020d98567e7a9f3cd8d8c8d37f1c5/sql/moz-fx-data-shared-prod/firefox_accounts_derived/fxa_users_daily_v1/query.sql#L55

This might be a left over from a very long time, as the way currently this model is structured that each time we include all previous records which results in user_id null entry to exist for each date:

```sql
SELECT MIN(first_seen_date)
FROM `firefox_accounts_derived.fxa_users_first_seen_v1`
WHERE
  first_seen_date >= "2021-01-01"
  AND user_id IS NULL
```
Returns: `2021-07-14`.

My guess is that ever since we see 1 entry per date with user_id being null since then because we just keep including that row due to how the logic is structured. I'm not sure if it's worth to re-compile the table since then though.

Reply:
Rebuilding this table from fxa_users_daily will cost ~$5 (1.11 TB) so I'd say it's worth it. Note that the init script still points to fxa_all_events which would process 7.91 TB.

Reply:
IMHO we should be using the daily table for building the first seen table. The question is whether we should keep the init.sql at all. If I'm correct in saying it our old pattern for initialising a table. The new approach is to have a schema.yaml and using our backfill process for populating the table.